stdlib/InteractiveUtils/test/runtests.jl

```diff
@@ -12,6 +12,57 @@ struct Type4Union end
 func4union(::Union{Type4Union,Int}) = ()
 @test !isempty(methodswith(Type4Union, @__MODULE__))
 
+module MWM
+export f, SubModule
+struct C end
+abstract type Super end
+struct Sub <: Super end
+abstract type SuperP{A<:AbstractArray} end
+abstract type SubP{A<:AbstractVector} <: SuperP{A} end
```

Comment:
Sadly, had I written `SubP{A} <: SuperP{A}` then the test would have failed. Presumably this is just #6383, but if there is a nice workaround I'd be happy to incorporate it. I am sure there's a lot of code out there that doesn't re-impose the same parameter bounds.